pkg/providers/cloudstack/controlplane.go

```diff
@@ -0,0 +1,129 @@
+package cloudstack
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	cloudstackv1 "sigs.k8s.io/cluster-api-provider-cloudstack/api/v1beta2"
+
+	"github.com/aws/eks-anywhere/pkg/clients/kubernetes"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/clusterapi"
+	yamlcapi "github.com/aws/eks-anywhere/pkg/clusterapi/yaml"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/yamlutil"
+)
+
+// BaseControlPlane represents a CAPI CloudStack control plane.
+type BaseControlPlane = clusterapi.ControlPlane[*cloudstackv1.CloudStackCluster, *cloudstackv1.CloudStackMachineTemplate]
+
+// ControlPlane holds the CloudStack specific objects for a CAPI CloudStack control plane.
+type ControlPlane struct {
+	BaseControlPlane
+	Secrets *corev1.Secret
```

Comment:
cloudstack cp doesn't have secret. https://github.com/aws/eks-anywhere/blob/main/pkg/providers/cloudstack/config/template-cp.yaml

Reply:
removed it